### PR TITLE
best model and parameter selection #7

### PR DESCRIPTION
**Pull request by muskan191103**
- Used OneHotEncoder for encoding 'type' column
- Replaced NaN values with 0 and mean value wherever required
- Used train_test_split method for dividing dataset in training and testing datasets
- Formed Dictionary with key value pairs as different models and their different parameters
- Using 'for' loop and GridSearchCV so that it can iterate over every combination possible for model selections given

### CODE REVIEW

**Comment by Samcoding5854** [comment]
Good work using gridsearchCV but you havent yet selected or declared the best model in the notebook
Do that only then I can accept this PR

**Comment by Samcoding5854** [comment]
Could have used and displayed the accuracies even if its taking time as the issue was to achieve optimal maximum accuracy. 
Rest implementation has no issues.
